### PR TITLE
[Mempool] AttachedPeer can be null when relaying.

### PR DESCRIPTION
**Pull request by fassadlr**
This is to see if this fixes the timeout in integration tests as this test fails to complete:

`MempoolReceiveFromManyNodes`

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs [hunk 1]
```diff
@@ -487,6 +487,12 @@ public void RelayTransaction(uint256 hash)
             IEnumerable<MempoolBehavior> behaviours = peers.Select(s => s.Behavior<MempoolBehavior>());
             foreach (MempoolBehavior mempoolBehavior in behaviours)
             {
+                if (mempoolBehavior?.AttachedPeer == null)
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: Do we have anything that cleans out old behaviours when they are no longer attached to anything?

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs [hunk 2]
```diff
@@ -487,6 +487,12 @@ public void RelayTransaction(uint256 hash)
...
+                {
+                    this.logger.LogTrace("Peer is null, skipped.");
+                    continue;
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: I don't think `continue;` is correct if we have no attached peer no point to continue.

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs (hunk 1 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: ah sorry this is per behaviour so ok its correct

File: src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs (hunk 2 above)
Comment (dangershony) [reply src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: you can still have race conditions later down the line,
I suggest taking a local var of the peer

```
var peer = mempoolBehavior?.AttachedPeer

if(peer == null)
  continue
```
form here us `peer`

Comment (fassadlr) [reply src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs]: Fixed 👍

**Pull request merged**